### DISCUSSION TITLE
Reference Kotlinpoet created class in another class.

### DISCUSSION THREAD

**Discussion by DeRealMorgan** [square/kotlinpoet#1846] [Q&A]
I did not find a way how I can  add a parameter in a constructor/cuntion parameter that should be generates by Kotlinpoet.

Either I dont understand the API or there is an obvious limitation or it is simply missing.

Kotlinpoet generates:
class X {}

How can I do this?: 
class Y(val x: X) {}

Thanks!

**Comment by JakeWharton** [comment]
You can use something like `ClassName("com.mattel", "Hoverboard")` to create a reference to a type that either doesn't exist yet, or isn't available to reference directly inside the code generator.

Here's a fairly self-contained example of using `ClassName` to reference types by name only:

https://github.com/square/kotlinpoet/blob/1b9e54bc89412fd5424f32a6e12aec05a72f8ce1/kotlinpoet/src/commonTest/kotlin/com/squareup/kotlinpoet/TypeSpecTest.kt#L1064-L1083
